src/tests/components/App.test.js

```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import App from "../../components/App";
+
+describe("App", () => {
+  it("renders correctly", () => {
+    const { asFragment } = render(<App />);
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it("renders Weather App header", () => {
+    render(<App />);
+    const appHeader = screen.getByText("Weather App");
+    expect(appHeader).toBeInTheDocument();
+  });
```

Comment:
When you render `App`, you make a call to the API. You do not want to actually make this call while testing. I would suggest spying on the `axios` `get` request and returning dummy data. this ends up looking like:
```
import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import axios from "axios";
import App from "../../components/App";
import data from "../../data/forecast.json";

describe("App", () => {
  beforeEach(() => {
    jest.spyOn(axios, "get").mockResolvedValue({ data });
  });

  it("renders correctly", async () => {
    const { asFragment } = render(<App />);

    await waitFor(() => {
      screen.getByText(`${data.location.city}, ${data.location.country}`)
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders data correctly", async () => {
    render(<App />);

    const appHeader = screen.getByText("Weather App");
    expect(appHeader).toBeInTheDocument();

    await waitFor(() => {
      const location = screen.getByText(
        `${data.location.city}, ${data.location.country}`
      );
      expect(location).toBeInTheDocument();
    });
  });
});
```

Reply:
I have put the `spyOn` in its own `beforeEach` to reduce code repetition. Then, in the tests I use `waitFor` to wait for the resolved data. In the snapshot, this means the snapshot actually contains the website with filled out data. In the other test, I can also check that the app renders the location data correctly after it gets the resolved data.